.github/workflows/pypi-publish.yml

```diff
@@ -25,11 +28,21 @@ jobs:
     permissions:
       id-token: "write"
     steps:
+      - name: Get publish-requirements.txt from repository
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          sparse-checkout: |
+            ${{ env.PUBLISH_REQUIREMENTS_PATH }}
+          sparse-checkout-cone-mode: false
+          ref: ${{ github.ref }}
```

Comment:
What ref is this going to be in practice? We trigger pypi-publish after a completed workflow run of wheel builder. Will this not always be tip of main in that case?

Reply:
Yes, I think it will always be main. The default for this parameter is actually very useful, as seen [here](https://github.com/actions/checkout), so I'll remove this line and keep it as the default.

A good reason I can think to use this line is if you want to trigger `workflow_dispatch` to other branch than main, but I don't think it will be the case.